app/models/order_details_updater.rb

```diff
@@ -12,7 +12,7 @@ def update
       zero_out
       add_new_details
     end
-
+    deduct_from_inventory if params[:order][:status] == "closed"
```

Comment:
I think this should happen in the order model itself, as part of the enum state change. Since we are using the [stateful_enum](https://github.com/amatsuda/stateful_enum) gem, I think it makes sense to put this as an after transition event in here: https://github.com/on-site/StockAid/blob/master/app/models/concerns/order_status.rb#L71